lib/JSX.ml

```diff
@@ -0,0 +1,112 @@
+module Attribute = struct
+  (** Used internally, no need to use *)
+
+  type t =
+    | Bool of (string * bool)
+    | String of (string * string)
+    | Style of string
+    | Event of (string * string)
+
+  let t_to_string attr =
+    match attr with
+    (* false attributes don't get rendered *)
+    | Bool (_, false) -> ""
+    (* true attributes render solely the attribute name *)
+    | Bool (k, true) -> k
+    | Style styles -> Printf.sprintf "style=\"%s\"" styles
+    | Event (name, value) -> Printf.sprintf "%s=\"%s\"" name value
+    | String (k, v) -> Printf.sprintf "%s=\"%s\"" k (Html.encode v)
+
+  let to_string attrs =
+    match List.map t_to_string attrs with
+    | [] -> ""
+    | rest -> " " ^ (rest |> List.rev |> String.concat " " |> String.trim)
+end
+
+type node = {
+  tag : string;
+  attributes : Attribute.t list;
+  children : element list;
+}
+
+and element =
+  | Null
+  | String of string
+  | Unsafe of string (* text without encoding *)
+  | Fragment of element list
+  | Node of node
+  | Component of (unit -> element) (* TODO: Is this unused? *) [@warning "-37"]
+  | List of element list
+
+let string txt = String txt
+let text = string
```

Comment:
Those alias, if people prefer it should go in their code

Reply:
Yep agree! It was more if we wanted to go from `text` to `string` and do it gracefully (deprecating `text`)

https://github.com/davesnx/html_of_jsx/pull/14/files#diff-7f3c56163c7254efc9a900b7d1b09fa9cc0eeb0e6057d30e1d2466d0cb73da44R67-R69